nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch.java

```diff
@@ -0,0 +1,326 @@
+        _rels.add(REL_HITS);
+        _rels.add(REL_AGGREGATIONS);
+        relationships = Collections.unmodifiableSet(_rels);
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(QUERY);
+        descriptors.add(QUERY_ATTRIBUTE);
+        descriptors.add(INDEX);
+        descriptors.add(TYPE);
+        descriptors.add(CLIENT_SERVICE);
+        descriptors.add(SPLIT_UP_HITS);
+        descriptors.add(SPLIT_UP_AGGREGATIONS);
+
+        propertyDescriptors = Collections.unmodifiableList(descriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        clientService = context.getProperty(CLIENT_SERVICE).asControllerService(ElasticSearchClientService.class);
+    }
+
+    @OnStopped
+    public void onStopped() {
+        this.clientService = null;
+    }
+
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    private String getQuery(FlowFile input, ProcessContext context, ProcessSession session) throws IOException {
+        String retVal = null;
+        if (context.getProperty(QUERY).isSet()) {
+            retVal = context.getProperty(QUERY).evaluateAttributeExpressions(input).getValue();
+        } else if (input != null) {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            session.exportTo(input, out);
+            out.close();
+
+            retVal = new String(out.toByteArray());
+        }
+
+        return retVal;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile input = null;
+        if (context.hasIncomingConnection()) {
+            input = session.get();
+
+            if (input == null && context.hasNonLoopConnection()) {
+                return;
+            }
+        }
+
+        try {
+
+            final String query = getQuery(input, context, session);
+            final String index = context.getProperty(INDEX).evaluateAttributeExpressions(input).getValue();
+            final String type = context.getProperty(TYPE).evaluateAttributeExpressions(input).getValue();
+
+            Optional<SearchResponse> resp = clientService.search(query, index, type);
+            if (resp.isPresent()) {
+                SearchResponse response = resp.get();
+
+                List<FlowFile> hitsFlowFiles = handleHits(response.getHits(), context, session, input);
+                List<FlowFile> aggsFlowFiles = handleAggregations(response.getAggregations(), context, session, input);
+
+                if (hitsFlowFiles.size() > 0) {
+                    for (FlowFile ff : hitsFlowFiles) {
+                        session.transfer(ff, REL_HITS);
```

Comment:
This should be able to be switched by to the single transfer since the root issue was setting the clientService to null on unScheduled.

Reply:
Done.